engine/chainstate/transaction_info.go

```diff
@@ -7,22 +7,17 @@ import (
 
 // TransactionInfo is the universal information about the transaction found from a chain provider
 type TransactionInfo struct {
-	BlockHash     string             `json:"block_hash,omitempty"`    // mAPI
-	BlockHeight   int64              `json:"block_height"`            // mAPI
-	Confirmations int64              `json:"confirmations,omitempty"` // mAPI
-	ID            string             `json:"id"`                      // Transaction ID (Hex)
-	MinerID       string             `json:"miner_id,omitempty"`      // mAPI ONLY - miner_id found
-	Provider      string             `json:"provider,omitempty"`      // Provider is our internal source
-	MerkleProof   *bc.MerkleProof    `json:"merkle_proof,omitempty"`  // mAPI 1.5 ONLY
-	BUMP          *bc.BUMP           `json:"bump,omitempty"`          // Arc
-	TxStatus      broadcast.TxStatus `json:"tx_status,omitempty"`     // Arc ONLY
+	BlockHash   string             `json:"block_hash,omitempty"` // Block hash of the transaction
+	BlockHeight int64              `json:"block_height"`         // Block height of the transaction
+	ID          string             `json:"id"`                   // Transaction ID (Hex)
+	Provider    string             `json:"provider,omitempty"`   // Provider is our internal source
+	BUMP        *bc.BUMP           `json:"bump,omitempty"`       // Merkle proof in BUMP format
+	TxStatus    broadcast.TxStatus `json:"tx_status,omitempty"`  // Status of the transaction
 }
 
-// Valid validates TransactionInfo by checking if it contains
-// BlockHash and MerkleProof (from mAPI) or BUMP (from Arc)
+// Valid validates TransactionInfo by checking if it contains BlockHash and BUMP
 func (t *TransactionInfo) Valid() bool {
 	arcInvalid := t.BUMP == nil
-	mApiInvalid := t.MerkleProof == nil || t.MerkleProof.TxOrID == "" || len(t.MerkleProof.Nodes) == 0
-	invalid := t.BlockHash == "" || (arcInvalid && mApiInvalid)
+	invalid := t.BlockHash == "" || arcInvalid
```

Comment:
It can be a "one-liner":
`return t.BlockHash != "" && t.BUMP != nil`

Reply:
reduced to a single line